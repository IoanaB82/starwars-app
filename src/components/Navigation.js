import React from "react";

import { NavLink } from "react-router-dom";

import styled from "styled-components";

const NavMenu = styled.nav`
  display: flex;
  justify-content: space-around;
  align-items: center;
`;

const Ul = styled.ul`
  list-style: none;
  text-decoration: none;
  color: #fff;
  display: flex;

  li {
    padding: 20px;

    & a,
    a:visited {
      text-decoration: none;
      color: #fff;
    }

    & a:active,
    a:focus {
      color: var(--color-light-brown);
    }

    & a:hover {
      color: #000;
    }
  }
`;

const Navigation = () => {
  return (
    <NavMenu>
      <Ul>
        <li>
          <NavLink to="/">Home</NavLink>
        </li>
        <li>
          <NavLink to="/species">Species</NavLink>
        </li>
        <li>
          <NavLink to="/planets">Planets</NavLink>
        </li>
        <li>
          <NavLink to="/favorites">Favorites</NavLink>
        </li>
      </Ul>
    </NavMenu>
  );
};

export default Navigation;
