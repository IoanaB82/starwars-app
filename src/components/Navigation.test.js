import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { MemoryRouter } from "react-router-dom";

import Navigation from "./Navigation";

afterEach(cleanup);

//test that component renders
it("renders without crashing", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Navigation />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

//it renders links
it("renders home link", () => {
  const { getByText } = render(
    <MemoryRouter>
      <Navigation />
    </MemoryRouter>
  );
  const linkElement = getByText("Home");
  expect(linkElement).toBeInTheDocument();
});

it("renders species link", () => {
  const { getByText } = render(
    <MemoryRouter>
      <Navigation />
    </MemoryRouter>
  );
  const linkElement = getByText("Species");
  expect(linkElement).toBeInTheDocument();
});

it("renders Planes link", () => {
  const { getByText } = render(
    <MemoryRouter>
      <Navigation />
    </MemoryRouter>
  );
  const linkElement = getByText("Planets");
  expect(linkElement).toBeInTheDocument();
});

it("renders Favorites link", () => {
  const { getByText } = render(
    <MemoryRouter>
      <Navigation />
    </MemoryRouter>
  );
  const linkElement = getByText("Favorites");
  expect(linkElement).toBeInTheDocument();
});

//nav item har correct link
it("species links to species", () => {
  const { getByText } = render(
    <MemoryRouter>
      <Navigation />
    </MemoryRouter>
  );
  const linkElement = getByText("Species");
  expect(linkElement.href).toBe("http://localhost/species");
});

it("species links to planets", () => {
  const { getByText } = render(
    <MemoryRouter>
      <Navigation />
    </MemoryRouter>
  );
  const linkElement = getByText("Planets");
  expect(linkElement.href).toBe("http://localhost/planets");
});
