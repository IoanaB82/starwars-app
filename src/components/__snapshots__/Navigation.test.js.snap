// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdnylx ktEToZ"
  >
    <ul
      class="sc-gtssRu hDxyLK"
    >
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/species"
        >
          Species
        </a>
      </li>
      <li>
        <a
          href="/planets"
        >
          Planets
        </a>
      </li>
      <li>
        <a
          href="/favorites"
        >
          Favorites
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
